### PR TITLE
[move source language] Actually compile && and ||

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Compiled && and || to if-else
- Kept the operator if the RHS is a value/local

## Test Plan

new tests

### CODE REVIEW

File: language/move-lang/tests/functional/evaluation_order/short_circuiting.move [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    true || X::error();
+    vtrue || X::error();
+    vtrue || { let r = X::error(); r };
+    { let x = vtrue; x} || X::error();
+
+    false && X::error();
+    vfalse && X::error();
+    vfalse && { let r = X::error(); r };
+    { let x = vfalse; x} && X::error();
+}
```
Comment (sblackshear) [inline language/move-lang/tests/functional/evaluation_order/short_circuiting.move]: Are there already tests to check that we 
- evaluate the RHS when the LHS is false for or/true for and
- evaluate the LHS in all cases (not that this is a likely mistake)?

Comment (tnowacki) [reply language/move-lang/tests/functional/evaluation_order/short_circuiting.move]: > evaluate the RHS when the LHS is false for or/true for and
> evaluate the LHS in all cases (not that this is a likely mistake)

I'll add them to be safe

**Pull request closed without merging**